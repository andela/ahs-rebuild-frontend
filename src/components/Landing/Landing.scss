.layout-body{
    // display: inline-flex;
    height: 735px;
    width: 100%;
    &-content{
        padding: 20px;
        margin-left: 30px;
        text-align: left;
    }
    &-tracks{
        display: flex;
        width: 50%;
        height: 20%;
        text-align: left;
        margin-left: 50px;
        &-desc{
            text-align: justify;
        }
    }
    &-intro{
        margin: 30px 20% 0px 20%;
        &-content{
            text-align: justify;
        }    
    }
}
.layout-history{
    height: 30px;
    background-color: #141f1f;
    width: 100%;
}
.layout-sidebar{
    height: 100%;
    width: 20%;
    background-color: #fff;
}
.beginner-track{
    position: absolute;
    right: 20%;
    &:hover{
        cursor: pointer;
    }
    
}
.advanced-track{
    position: absolute;
    right: 10%;
    &:hover{
        cursor: pointer;
    }

}
