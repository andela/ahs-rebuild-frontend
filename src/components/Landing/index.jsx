import React, { Component, Fragment } from 'react';
import Navbar from '../Navbar/index.jsx';
import Button from '../Buttons';
import './Landing.scss';


class Landing extends Component {
    
    handleChangeTrack = event => {
        event.preventDefault();
        
      };
    render(){
        return(
            <Fragment>
                <Navbar />
                <div className="layout-history">Jump to:</div>
                <div className="layout-body">
                    <div className="layout-body-content">
                        <h1> Overview </h1>
                    </div>
                    <div className="layout-body-tracks">
                        <div className='layout-body-tracks-desc'>
                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. 
                            Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, 
                            when an unknown printer took a galley of type and scrambled it to make a type specimen book. 
                            It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
                            It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, 
                            and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                        </div>
                        <Button
                            buttonName="beginnerTrack"
                            handleChange={this.handleChangeTrack}
                            buttonClassName='beginner-track'
                            track='Beginner Track'
                        />
                        <Button
                            buttonName="advancedTrack"
                            handleChange={this.handleChangeTrack}
                            buttonClassName='advanced-track'
                            track='Advanced Track'
                        />
                    </div>
                    <div className='layout-body-intro'>
                        <h2>Introduction View</h2>
                        <div className='layout-body-intro-content'> 
                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. 
                            Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, 
                            when an unknown printer took a galley of type and scrambled it to make a type specimen book. 
                        </div>
                    </div>
                </div>
            </Fragment>
        )
    }
}

export default Landing;
