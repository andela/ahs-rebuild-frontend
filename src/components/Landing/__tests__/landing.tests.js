import React from 'react';
import { shallow } from 'enzyme'
import Landing from '../../Landing';


describe('Test Landing', () => {
    it('Renders as expected', () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper).toMatchSnapshot()
    })
})
