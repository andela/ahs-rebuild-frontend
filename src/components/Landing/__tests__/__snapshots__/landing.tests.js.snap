// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Landing Renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Navbar />,
        <div
          className="layout-history"
        >
          Jump to:
        </div>,
        <div
          className="layout-body"
        >
          <div
            className="layout-body-content"
          >
            <h1>
               Overview 
            </h1>
          </div>
          <div
            className="layout-body-tracks"
          >
            <div
              className="layout-body-tracks-desc"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
            </div>
            <Button
              buttonClassName="beginner-track"
              buttonName="beginnerTrack"
              handleChange={[Function]}
              track="Beginner Track"
            />
            <Button
              buttonClassName="advanced-track"
              buttonName="advancedTrack"
              handleChange={[Function]}
              track="Advanced Track"
            />
          </div>
          <div
            className="layout-body-intro"
          >
            <h2>
              Introduction View
            </h2>
            <div
              className="layout-body-intro-content"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jump to:",
          "className": "layout-history",
        },
        "ref": null,
        "rendered": "Jump to:",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="layout-body-content"
            >
              <h1>
                 Overview 
              </h1>
            </div>,
            <div
              className="layout-body-tracks"
            >
              <div
                className="layout-body-tracks-desc"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
              </div>
              <Button
                buttonClassName="beginner-track"
                buttonName="beginnerTrack"
                handleChange={[Function]}
                track="Beginner Track"
              />
              <Button
                buttonClassName="advanced-track"
                buttonName="advancedTrack"
                handleChange={[Function]}
                track="Advanced Track"
              />
            </div>,
            <div
              className="layout-body-intro"
            >
              <h2>
                Introduction View
              </h2>
              <div
                className="layout-body-intro-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
              </div>
            </div>,
          ],
          "className": "layout-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                 Overview 
              </h1>,
              "className": "layout-body-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Overview ",
              },
              "ref": null,
              "rendered": " Overview ",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="layout-body-tracks-desc"
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                </div>,
                <Button
                  buttonClassName="beginner-track"
                  buttonName="beginnerTrack"
                  handleChange={[Function]}
                  track="Beginner Track"
                />,
                <Button
                  buttonClassName="advanced-track"
                  buttonName="advancedTrack"
                  handleChange={[Function]}
                  track="Advanced Track"
                />,
              ],
              "className": "layout-body-tracks",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                  "className": "layout-body-tracks-desc",
                },
                "ref": null,
                "rendered": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonClassName": "beginner-track",
                  "buttonName": "beginnerTrack",
                  "handleChange": [Function],
                  "track": "Beginner Track",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonClassName": "advanced-track",
                  "buttonName": "advancedTrack",
                  "handleChange": [Function],
                  "track": "Advanced Track",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Introduction View
                </h2>,
                <div
                  className="layout-body-intro-content"
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
                </div>,
              ],
              "className": "layout-body-intro",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Introduction View",
                },
                "ref": null,
                "rendered": "Introduction View",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
                  "className": "layout-body-intro-content",
                },
                "ref": null,
                "rendered": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Navbar />,
          <div
            className="layout-history"
          >
            Jump to:
          </div>,
          <div
            className="layout-body"
          >
            <div
              className="layout-body-content"
            >
              <h1>
                 Overview 
              </h1>
            </div>
            <div
              className="layout-body-tracks"
            >
              <div
                className="layout-body-tracks-desc"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
              </div>
              <Button
                buttonClassName="beginner-track"
                buttonName="beginnerTrack"
                handleChange={[Function]}
                track="Beginner Track"
              />
              <Button
                buttonClassName="advanced-track"
                buttonName="advancedTrack"
                handleChange={[Function]}
                track="Advanced Track"
              />
            </div>
            <div
              className="layout-body-intro"
            >
              <h2>
                Introduction View
              </h2>
              <div
                className="layout-body-intro-content"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jump to:",
            "className": "layout-history",
          },
          "ref": null,
          "rendered": "Jump to:",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="layout-body-content"
              >
                <h1>
                   Overview 
                </h1>
              </div>,
              <div
                className="layout-body-tracks"
              >
                <div
                  className="layout-body-tracks-desc"
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                </div>
                <Button
                  buttonClassName="beginner-track"
                  buttonName="beginnerTrack"
                  handleChange={[Function]}
                  track="Beginner Track"
                />
                <Button
                  buttonClassName="advanced-track"
                  buttonName="advancedTrack"
                  handleChange={[Function]}
                  track="Advanced Track"
                />
              </div>,
              <div
                className="layout-body-intro"
              >
                <h2>
                  Introduction View
                </h2>
                <div
                  className="layout-body-intro-content"
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
                </div>
              </div>,
            ],
            "className": "layout-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                   Overview 
                </h1>,
                "className": "layout-body-content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Overview ",
                },
                "ref": null,
                "rendered": " Overview ",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="layout-body-tracks-desc"
                  >
                    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                  </div>,
                  <Button
                    buttonClassName="beginner-track"
                    buttonName="beginnerTrack"
                    handleChange={[Function]}
                    track="Beginner Track"
                  />,
                  <Button
                    buttonClassName="advanced-track"
                    buttonName="advancedTrack"
                    handleChange={[Function]}
                    track="Advanced Track"
                  />,
                ],
                "className": "layout-body-tracks",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                    "className": "layout-body-tracks-desc",
                  },
                  "ref": null,
                  "rendered": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonClassName": "beginner-track",
                    "buttonName": "beginnerTrack",
                    "handleChange": [Function],
                    "track": "Beginner Track",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonClassName": "advanced-track",
                    "buttonName": "advancedTrack",
                    "handleChange": [Function],
                    "track": "Advanced Track",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Introduction View
                  </h2>,
                  <div
                    className="layout-body-intro-content"
                  >
                    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
                  </div>,
                ],
                "className": "layout-body-intro",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Introduction View",
                  },
                  "ref": null,
                  "rendered": "Introduction View",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
                    "className": "layout-body-intro-content",
                  },
                  "ref": null,
                  "rendered": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
